superset/assets/javascripts/explorev2/reducers/exploreReducer.js

```diff
@@ -90,6 +87,7 @@ export const exploreReducer = function (state, action) {
           chartUpdateEndTime: null,
           chartUpdateStartTime: now(),
           triggerQuery: false,
+          queryRequest: action.queryRequest,
```

Comment:
what about keeping the object in the component's state as `state.latestQueryRequest` and aborting it (if it exists) before setting the new one in the state